script.js

```diff
@@ -1,4 +1,124 @@
+    return "Welcome. Please enter the number of players for this game";
+
+    // in this mode, user enters number of players. game mode switches to dice roll mode.
+  } else if (gameMode == ENTERNUMOFUSERS) {
+    numOfPlayers = input;
+    console.log("store no of players " + numOfPlayers);
+    gameMode = DICEROLLMODE;
+    return (
+      "There are " +
+      numOfPlayers +
+      " players in this game. Now, player 1 please click submit twice to roll two dice."
+    );
+  }
+  // entering dice roll mode. The loop ensures one game per user, with each game entails two dice rolls and the ordering of the dice rolls.
+
+  while (numOfUsersWhoHavePlayed < numOfPlayers) {
+    console.log(numOfUsersWhoHavePlayed, numOfPlayers);
+
+    // include one more round on top of max dice rolls allowed. in the first two round user rolls dice, in last round user orders the dice rolls
+    while (diceRollsPerUser < diceRollsAllowedPerUser + 1) {
+      //the following while loop rolls dice the num of times allowed per user, in this case 2.
+      while (diceRollsPerUser < diceRollsAllowedPerUser) {
+        var showAndStoreDiceRolls = rollAndStoreDice();
+        diceRollsPerUser = diceRollsPerUser + 1;
+        numDiceRollsOverall = numDiceRollsOverall + 1;
+        return (
+          "The dice rolled is " + showAndStoreDiceRolls + ". Click submit."
+        );
+      }
+      // after dice rolls exceeds dice rolls allowed per user (2), enter order mode.
+
+      if (gameMode == DICEROLLMODE) {
+        gameMode = ORDERMODE;
+        var guess1 = allDiceRollsArray[numDiceRollsOverall - 1];
+        var guess2 = allDiceRollsArray[numDiceRollsOverall - 2];
+        return (
+          "Your guesses are " +
+          guess1 +
+          " and " +
+          guess2 +
+          ". Order the two guesses to form a two-digit number. The player with the highest two-digit number wins."
+        );
+      } else if (gameMode == ORDERMODE) {
+        console.log("entered order mode");
+
+        orderedNumberArray.push(input);
+        orderedNumberArrayforSorting.push(input);
+        gameMode = DICEROLLMODE;
+        numOfUsersWhoHavePlayed = numOfUsersWhoHavePlayed + 1;
+        diceRollsPerUser = diceRollsPerUser + 1;
+
+        if (playerNumber == numOfPlayers) {
+          return (
+            "You keyed in " +
+            orderedNumberArray[numOfUsersWhoHavePlayed - 1] +
+            ". All " +
+            numOfPlayers +
+            " players have gone a round. Now click submit to see who won!"
+          );
+        } else
+          return (
+            "You keyed in " +
+            orderedNumberArray[numOfUsersWhoHavePlayed - 1] +
+            ". It is now the next player's turn"
+          );
+      }
+    }
+    diceRollsPerUser = 0;
+    playerNumber = playerNumber + 1;
+    return "Hello player no. " + playerNumber + " click submit to play.";
+  }
+  return outputWinner(orderedNumberArrayforSorting, orderedNumberArray);
```

Comment:
To allow your game to start a new round without refreshing the browser, you might need to reinitialise some variables (`numOfUsersWhoHavePlayer`, `playerNumber`) to `0` to meet the while conditions